cmd/benchsort/main.go

```diff
@@ -0,0 +1,438 @@
+			}
+			log.Fatal(err)
+		}
+
+		cLog("Generating...")
+		start := time.Now()
+		if err := export(); err != nil {
+			log.Fatal(err)
+		}
+		cLog("Done!")
+		cLogf("Data placed in: %s", path.Join(tmpDir, "data.out"))
+		cLog("Time used: ", time.Since(start))
+		cLog("=================================")
+	}
+
+	if runCmd.Parsed() {
+		// Sanity checks
+		if bufSize <= 0 {
+			log.Fatal(errors.New("buffer size must be positive"))
+		}
+		if inputRatio < 0 || inputRatio > 100 {
+			log.Fatal(errors.New("input ratio must between 0 and 100 (inclusive)"))
+		}
+		if outputRatio < 0 || outputRatio > 100 {
+			log.Fatal(errors.New("output ratio must between 0 and 100 (inclusive)"))
+		}
+		if _, err := os.Stat(tmpDir); err != nil {
+			if os.IsNotExist(err) {
+				log.Fatal(errors.New("tmpDir does not exist"))
+			}
+			log.Fatal(err)
+		}
+
+		var (
+			err     error
+			dir     string
+			data    []*comparableRow
+			profile *os.File
+			fs      *filesort.FileSorter
+		)
+		cLog("Loading...")
+		start := time.Now()
+		data, err = load(inputRatio)
+		if err != nil {
+			log.Fatal(err)
+		}
+		cLog("Done!")
+		cLogf("Loaded %d rows", len(data))
+		cLog("Time used: ", time.Since(start))
+		cLog("=================================")
+
+		sc := new(variable.StatementContext)
+		fsBuilder := new(filesort.Builder)
+		byDesc := make([]bool, keySize)
+		for i := 0; i < keySize; i++ {
+			byDesc[i] = false
+		}
+		dir, err = ioutil.TempDir(tmpDir, "benchsort_test")
+		if err != nil {
+			log.Fatal(err)
+		}
+		fs, err = fsBuilder.SetSC(sc).SetSchema(keySize, valSize).SetBuf(bufSize).SetDesc(byDesc).SetDir(dir).Build()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		if cpuprofile != "" {
+			profile, err = os.Create(cpuprofile)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+
+		cLog("Inputing...")
+		start = time.Now()
+		for _, r := range data {
+			err = fs.Input(r.key, r.val, r.handle)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		cLog("Done!")
+		cLogf("Input %d rows", len(data))
+		cLog("Time used: ", time.Since(start))
+		cLog("=================================")
+
+		cLog("Outputing...")
+		totalRows := int(float64(len(data)) * (float64(outputRatio) / 100.0))
+		start = time.Now()
+		if cpuprofile != "" {
+			pprof.StartCPUProfile(profile)
+		}
+		for i := 0; i < totalRows; i++ {
+			_, _, _, err = fs.Output()
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		if cpuprofile != "" {
+			pprof.StopCPUProfile()
+		}
+		cLog("Done!")
+		cLogf("Output %d rows", totalRows)
+		cLog("Time used: ", time.Since(start))
+		cLog("=================================")
+
+		cLog("Closing...")
+		start = time.Now()
+		err = fs.Close()
+		if err != nil {
+			log.Fatal(err)
+		}
+		cLog("Done!")
+		cLog("Time used: ", time.Since(start))
+		cLog("=================================")
+	}
+}
+
+func cLogf(format string, args ...interface{}) {
+	str := fmt.Sprintf(format, args...)
+	fmt.Println("\033[0;32m" + str + "\033[0m")
```

Comment:
I am trying to follow the convention from `cmd/benchdb`, it uses customized `cLogf`.